dynamiqs/time_array.py

```diff
@@ -138,6 +168,16 @@ def timecallable(f: callable[[float], Array]) -> CallableTimeArray:
     Returns:
        _(time-array object of shape (..., n, n) when called)_ Callable object
             returning $O(t)$ for any time $t$.
+
+    Examples:
+        >>> f = lambda t: jnp.cos(jnp.pi * t) * dq.sigmaz()
+        >>> H = dq.timecallable(f)
+        >>> H(0.0)
+        Array([[ 1.+0.j,  0.+0.j],
+               [ 0.+0.j, -1.+0.j]], dtype=complex64)
+        >>> H(1.0)
+        Array([[-1.+0.j, -0.+0.j],
+               [-0.+0.j,  1.-0.j]], dtype=complex64)
```

Comment:
We should **never** give an example for timecallable that is actually doable with `dq.modulated`, this is misleading on what `timecallable` should be used for. We can for example us the same example as the one in time-dependent operators.